classes/db/DAO.php

```diff
@@ -330,7 +330,7 @@ public function convertFromDB($value, $type, $nullable = false)
                 return (float) $value;
             case 'object':
             case 'array':
-                $decodedValue = json_decode($value, true);
+                $decodedValue = json_decode($value ?? ($type === 'array' ? '[]' : '{}'), true);
```

Comment:
AFAICR I got this from a warning in a user log.

A not null field is attempting to convert a null value. I thought it's better to return an empty array/object than `false`.